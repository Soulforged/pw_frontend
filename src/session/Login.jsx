//@flow
import React from "react";
import { Redirect } from "react-router-dom";
import { compose, withState, withHandlers } from "recompose";

import type { Session, Location } from "src/types";

type Props = {
  ...Session,
  onChangeUsername: (string) => void,
  onChangePassword: (string) => void,
  onSubmit: (Event) => void,
  location: Location,
  username: string,
  password: string,
  login: (Object) => void
};

const component = (props: Props) => {
  const { from } = props.location.state || { from: { pathname: "/" } };
  if (props.user) {
    return <Redirect to={from} />;
  }
  return (
    <div id="login-wrapper">
      <div id="login-frame">
        <div id="login" className="pnls">
          <div className="login-header">
            <p className="login-title color-6">BACK OFFICE</p>
          </div>
          <form id="login_frm" onSubmit={props.onSubmit}>
            <div className="login-body">
              <div className="modal-p">
                <div className="input-group">
                  <span className="input-group-addon">
                    <span className="login-user" />
                  </span>
                  <input
                    type="text"
                    className="form-control"
                    id="usrnm"
                    placeholder="User@domain.com"
                    required
                    onChange={props.onChangeUsername}
                  />
                </div>
              </div>
              <div className="modal-p">
                <div className="input-group">
                  <span className="input-group-addon">
                    <span className="login-pswd" />
                  </span>
                  <input
                    type="password"
                    className="form-control"
                    id="pswd"
                    placeholder="Password"
                    onChange={props.onChangePassword}
                  />
                </div>
              </div>
            </div>
            <div className="login-footer">
              <div className="modal-p">
                <input type="submit" value="LOGIN" className="btn btn-md" />
              </div>
              <div className="modal-p">
                <a href="/forgotPassword" className="forgot-pswd color-6">Forgot your password?</a>
              </div>
              <div className="row color-6">
                <div className="col-md-6 col-sm-6 text-left">Powered by Peg b</div>
                <div className="col-md-6 col-sm-6 text-right">
                  <i className="logo-s" />
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default compose(
  withState("username", "setUsername", ""),
  withState("password", "setPassword", ""),
  withHandlers({
    onChangeUsername: props => (event) => {
      event.preventDefault();
      props.setUsername(event.target.value);
    },
    onChangePassword: props => (event) => {
      event.preventDefault();
      props.setPassword(event.target.value);
    },
    onSubmit: (props: Props) => (event) => {
      event.preventDefault();
      const { username, password } = props;
      props.login({ username, password });
    }
  })
)(component);
