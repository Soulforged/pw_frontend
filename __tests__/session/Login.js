//@flow
import React from "react";
import Login from "src/session/Login";
import renderer from "react-test-renderer";
import { Router } from "react-router";
import { mount } from "enzyme";
import createHistory from "history/createBrowserHistory";

it("login matches snapshot", () => {
  const rendered = renderer.create(<Login location={{}} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("login with user in session shows home", () => {
  const render = mount((
    <Router history={createHistory()}>
      <Login location={{ state: { from: "/other" } }} user={{}} />
    </Router>
  ));
  expect(render.props().history.location.pathname).toEqual("/other");
});

describe("login control test", () => {
  const login = jest.fn();
  const wrapper = mount(<Login location={{}} login={login} />);
  const form = wrapper.find("form");
  const formProps = form.props();

  it("calls login with valid form", () => {
    const usernameInput = wrapper.find("input#username");
    const passwordInput = wrapper.find("input#password");
    usernameInput.simulate("change", { target: { value: "username" } });
    passwordInput.simulate("change", { target: { value: "password" } });
    formProps.onSubmit({ preventDefault: jest.fn() });
    expect(login).toHaveBeenCalledWith({ username: "username", password: "password" });
  });
});
