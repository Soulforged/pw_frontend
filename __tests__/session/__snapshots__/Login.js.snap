// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login matches snapshot 1`] = `
<div
  id="login-wrapper"
>
  <div
    id="login-frame"
  >
    <div
      className="pnls"
      id="login"
    >
      <div
        className="login-header"
      >
        <p
          className="login-title color-6"
        >
          BACK OFFICE
        </p>
      </div>
      <form
        id="login_frm"
        onSubmit={[Function]}
      >
        <div
          className="login-body"
        >
          <div
            className="modal-p"
          >
            <div
              className="input-group"
            >
              <span
                className="input-group-addon"
              >
                <span
                  className="login-user"
                />
              </span>
              <input
                className="form-control"
                id="username"
                onChange={[Function]}
                placeholder="User@domain.com"
                required={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="modal-p"
          >
            <div
              className="input-group"
            >
              <span
                className="input-group-addon"
              >
                <span
                  className="login-pswd"
                />
              </span>
              <input
                className="form-control"
                id="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
        </div>
        <div
          className="login-footer"
        >
          <div
            className="modal-p"
          >
            <input
              className="btn btn-md"
              type="submit"
              value="LOGIN"
            />
          </div>
          <div
            className="modal-p"
          >
            <a
              className="forgot-pswd color-6"
              href="/forgotPassword"
            >
              Forgot your password?
            </a>
          </div>
          <div
            className="row color-6"
          >
            <div
              className="col-md-6 col-sm-6 text-left"
            >
              Powered by Peg b
            </div>
            <div
              className="col-md-6 col-sm-6 text-right"
            >
              <i
                className="logo-s"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
